commonknowledge/meep-intelligence-hub
# Tileserver

- Optimise analytical queries
- Deliver geojson points to the map via a dynamic tileserver endpoint
- Generalises the permissions calculator for use in the above, and adds tests for them too

## Motivation and Context

Analytical queries were taking a long time, so we:
- Split them up into separate queries
- Cut down their speed by 50%, using some python profiler tools to identify optimisations.

There was also no reason to load all 50k geojson points in a dataset, when you can only visually see a few at a time at low zoom levels, so this was handed off to a tileserver.

## How Can It Be Tested?

- Load a map and decide whether it's faster or not.
- Zoom in and see the geojson points.
- Run `python manage.py test hub.tests.test_permissions` to confirm that the right people can see the right things.
